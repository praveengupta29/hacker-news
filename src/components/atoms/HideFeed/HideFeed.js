import React from 'react';
import PropTypes from 'prop-types';

import './HideFeed.scss';
import Anchor from '../Anchor';

const HideFeed = ({ hideFeed, objectId }) => {
  return (
    <Anchor
      className="hide-btn"
      handleLinkClick={e => {
        e.preventDefault();
        hideFeed(objectId);
      }}
    >
      hide
    </Anchor>
  );
};

HideFeed.propTypes = {
  hideFeed: PropTypes.func.isRequired,
  objectId: PropTypes.string.isRequired,
};

export default HideFeed;
