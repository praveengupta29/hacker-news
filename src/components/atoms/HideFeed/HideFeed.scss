.hide-btn {
  margin: 0 4px;
}
