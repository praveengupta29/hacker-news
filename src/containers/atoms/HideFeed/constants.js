export const HIDE_FEED_KEY = 'HACKER_NEWS/FEED_LIST/HIDE_FEED_KEY';
export const HIDE_FEED = 'HACKER_NEWS/FEED_LIST/HIDE_FEED';
