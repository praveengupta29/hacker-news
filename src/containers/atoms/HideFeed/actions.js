import { HIDE_FEED } from './constants';

export const hideFeed = data => ({
  type: HIDE_FEED,
  data,
});
