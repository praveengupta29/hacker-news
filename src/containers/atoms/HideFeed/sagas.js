import { takeLatest } from 'redux-saga/effects';

import { HIDE_FEED, HIDE_FEED_KEY } from './constants';
import { LocalStorageUtil } from '../../../utils/localStorage';

export function* saveHideFeedSaga(action) {
  const { data: objectId } = action;

  const storage = new LocalStorageUtil();

  storage.saveItemInArray(HIDE_FEED_KEY, objectId);

  // Here, a API request can be made to hide feed
}

export default function* hideFeedSaga() {
  yield takeLatest(HIDE_FEED, saveHideFeedSaga);
}
