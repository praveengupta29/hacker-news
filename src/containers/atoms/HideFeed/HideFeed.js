import { connect } from 'react-redux';

import { hideFeed } from './actions';

import HideFeed from '../../../components/atoms/HideFeed';

export const mapDispatchToProps = dispatch => {
  return {
    hideFeed: feedId => {
      dispatch(hideFeed(feedId));
    },
  };
};

export default connect(
  null,
  mapDispatchToProps,
)(HideFeed);
