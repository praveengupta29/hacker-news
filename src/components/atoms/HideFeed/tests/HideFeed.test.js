import React from 'react';
import { shallow } from 'enzyme';
import HideFeed from '../HideFeed';

describe('HideFeed component', () => {
  const props = {
    hideFeed: () => {},
    objectId: '635483',
  };

  it('should render correctly and match snapshot', () => {
    const component = shallow(<HideFeed {...props} />);
    expect(component).toMatchSnapshot();
  });
});
