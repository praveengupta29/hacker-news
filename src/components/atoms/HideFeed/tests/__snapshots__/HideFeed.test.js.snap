// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HideFeed component should render correctly and match snapshot 1`] = `
<Anchor
  className="hide-btn"
  handleLinkClick={[Function]}
  target="_self"
  to=""
>
  hide
</Anchor>
`;
